core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -333,22 +333,25 @@ object Extraction {
 
     def newTuple(root: JValue, mappings: List[Mapping]): Any = {
       root match {
-        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+        case JArray(items) if items.nonEmpty && items.length <= tuples.length =>
           val builtItems: Seq[Object] = items.zip(mappings).map({
-            case (i,m) =>
-              build(i, m).asInstanceOf[Object]
+            case (item, mapping) =>
+              build(item, mapping).asInstanceOf[Object]
           })
-          val tupleClass = Class.forName("scala.Tuple" + items.length)
+          val tupleIndex = items.length - 1
+          val tupleClass = tuples.lift(tupleIndex).getOrElse {
+            throw new IllegalArgumentException(s"Cannot instantiate a tuple of length ${items.length} even though that should be a valid tuple length.")
+          }
 
           val typedTupleConstructor = tupleClass.getConstructors()(0)
           typedTupleConstructor.newInstance(builtItems: _*)
 
-        case JObject(items) if items.forall(_.name.startsWith("_")) =>
-          newTuple(JArray(items.map(_.value)), mappings)
-
         case JArray(items) =>
           throw new IllegalArgumentException("Cannot create a tuple of length " + items.length)
 
+        case JObject(items) if items.forall(_.name.startsWith("_")) =>
+          newTuple(JArray(items.map(_.value)), mappings)
```

Comment:
Still wondering about this. Even if we have a backward-compat deal that shows that we used to serialize `_1` and friends in order, the old way of dealing with tuples (which is constructor-based) would have dealt with a user furnishing an out-of-order set of _*s (not necessarily serialized from here). If we have a test that shows that you can pass `{"_3": "5", "_1": "1", "_2": "8"}` (e.g.) to a `(String, String, String)` and it deserializes the same way with this extraction strategy, I'm game. But I don't see how that would happen, tbh.